packages/core/src/queries/application.ts

```diff
@@ -47,6 +54,43 @@ export const createApplicationQueries = (pool: CommonQueryMethods) => {
     return { count: Number(count) };
   };
 
+  const countM2mApplicationsByIds = async (applicationIds: string[]) => {
+    const { count } = await pool.one<{ count: string }>(sql`
+      select count(*)
+      from ${table}
+      where ${fields.type} = ${ApplicationType.MachineToMachine}
+      ${conditionalSql(
+        applicationIds,
```

Comment:
should throw error if ids is empty?

Reply:
Good call, I think can directly return the result if the `ids` is empty.

Reply:
But it seems that this code is deleted:
```
if (applicationIds.length === 0) {
  return { count: 0 };
}
```

Reply:
Forgot to push the latest changes, it should be good now.